src/backends/gerrit.js

```diff
@@ -37,54 +40,46 @@ export class GerritBackend {
     return fetch(changesUrl, {credentials: 'include'})
       .then((res) => res.text())
       .then((text) => {
-        return JSON.parse(text.substring(text.indexOf('\n') + 1));
-      }).then((data) => {
+        let data = JSON.parse(text.substring(text.indexOf('\n') + 1))
         return this.parseResponse_(data, selfAddress);
       });
   }
 
   ensureLogin_() {
-    return this.getSelfAddress_().then((selfAddress) => {
-      return selfAddress;
-    }, (err) => {
-      return this.login_().then(this.getSelfAddress_.bind(this));
-    });
+    return this.getSelfAddress_().catch((err) =>
+	this.login_().then(() => this.getSelfAddress_()));
   }
 
   getSelfAddress_() {
     const accountsUrl = this.site_['url'] + '/accounts/self';
     return fetch(accountsUrl, {credentials: 'include'})
       .then((res) => res.text())
-      .then((text) => {
-        return JSON.parse(text.substring(text.indexOf('\n') + 1));
-      }).then((data) => data.email);
+      .then((text) =>
+        JSON.parse(text.substring(text.indexOf('\n') + 1)).email);
   }
 
   login_() {
-    return new Promise((resolve, reject) => {
-      const loginUrl = this.site_['url'] + '/login/';
-      chrome.tabs.create({url: loginUrl, active: true}, (tab) => {
+    const loginUrl = this.site_['url'] + '/login/';
+    return chromeAsync.tabs.create({url: loginUrl, active: true})
```

Comment:
In the current implementation, the promise returned from login_() is resolved once login finishes. But IIUC in the new implementation it is resolved as soon as a new tab is created?